src/core/observables/utils_histogram.hpp

```diff
@@ -21,10 +21,36 @@
 
 #include <utils/Histogram.hpp>
 
+#include <boost/mpi/collectives/gather.hpp>
+#include <boost/mpi/communicator.hpp>
+#include <boost/serialization/vector.hpp>
+
 #include <cstddef>
+#include <utility>
 #include <vector>
 
-namespace Observables {
+namespace Observables::detail {
+
+/** @brief Gather data from all MPI ranks. */
+template <class Pos>
+auto gather(boost::mpi::communicator const &comm,
+            std::vector<Pos> const &local_pos) {
+  std::vector<std::vector<Pos>> global_pos{};
```

Comment:
```suggestion
  std::vector<std::vector<Pos>> global_pos{};
  global_pos.reserve(comm.size());
```

Reply:
my bad. Getting gcc-10 to properly capture line coverage information for `std::vector` constructors took some trial and error. Let's add those `.reserve()` back.